project-base/app/src/Controller/Front/SocialNetworkController.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\Front;
+
+use App\Model\Customer\User\RegistrationDataFactory;
+use App\Model\Customer\User\RegistrationFacade;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Security\Authenticator;
+use Shopsys\FrameworkBundle\Model\SocialNetwork\Exception\SocialNetworkLoginException;
+use Shopsys\FrameworkBundle\Model\SocialNetwork\SocialNetworkConfigFactory;
+use Shopsys\FrameworkBundle\Model\SocialNetwork\SocialNetworkFacade;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+class SocialNetworkController extends AbstractController
+{
+    protected const string REFERER_URL = 'refererUrl';
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Security\Authenticator $authenticator
+     * @param \Shopsys\FrameworkBundle\Model\SocialNetwork\SocialNetworkConfigFactory $socialNetworkConfigFactory
+     * @param \App\Model\Customer\User\RegistrationFacade $registrationFacade
+     * @param \App\Model\Customer\User\RegistrationDataFactory $registrationDataFactory
+     * @param \Shopsys\FrameworkBundle\Model\SocialNetwork\SocialNetworkFacade $socialNetworkFacade
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        protected readonly Authenticator $authenticator,
+        protected readonly SocialNetworkConfigFactory $socialNetworkConfigFactory,
+        protected readonly RegistrationFacade $registrationFacade,
+        protected readonly RegistrationDataFactory $registrationDataFactory,
+        protected readonly SocialNetworkFacade $socialNetworkFacade,
+        protected readonly Domain $domain,
+    ) {
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param string $type
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function loginAction(Request $request, string $type): Response
+    {
+        if (in_array($type, SocialNetworkConfigFactory::ALLOWED_PROVIDERS, true) === false) {
```

Comment:
Why are you checking types agains some hard defined array instead of checking if type is in `social_network_login_config` configuration? 

I would like to see that if you want to add for example Apple ID, you can do that just by configuration (and probably you will have to add some images to SF) but thats all. I don't like the idea when you have configuration for some service and then you have same information somewhere in codebase